sharding-proxy/sharding-proxy-backend/src/main/java/org/apache/shardingsphere/shardingproxy/backend/response/query/QueryHeader.java

```diff
@@ -48,20 +48,44 @@
     private final Integer columnType;
     
     private final int decimals;
+
+    private final boolean signed;
+
+    private final boolean primaryKey;
+
+    private final boolean nullable;
+
+    private final boolean autoIncrement;
     
     public QueryHeader(final ResultSetMetaData resultSetMetaData, final LogicSchema logicSchema, final int columnIndex) throws SQLException {
         this.schema = logicSchema.getName();
         if (logicSchema instanceof ShardingSchema) {
             Collection<String> tableNames = logicSchema.getShardingRule().getLogicTableNames(resultSetMetaData.getTableName(columnIndex));
             this.table = tableNames.isEmpty() ? "" : tableNames.iterator().next();
+            if (resultSetMetaData.getTableName(columnIndex).isEmpty()) {
+                this.primaryKey = false;
+                this.nullable = false;
+                this.autoIncrement = false;
+            } else {
+                this.primaryKey = logicSchema.getMetaData().getTables().get(resultSetMetaData.getTableName(columnIndex)).getColumns()
+                        .get(resultSetMetaData.getColumnName(columnIndex).toLowerCase()).isPrimaryKey();
+                this.nullable = logicSchema.getMetaData().getTables().get(resultSetMetaData.getTableName(columnIndex)).getColumns()
+                        .get(resultSetMetaData.getColumnName(columnIndex).toLowerCase()).isNotNull();
```

Comment:
How could 'isNotNull' get the right value if its value is 'nullable'?

Reply:
Unified name is notNull.